src/App.tsx

```diff
@@ -1,14 +1,64 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getAllTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Filter, FilterField } from './types/Filter';
+
+function getFilteredTodos(todos :Todo[], filter: Filter) :Todo[] {
+  let newTodos = [...todos];
+
+  newTodos = newTodos.filter(todo => {
+    switch (filter.field) {
+      case FilterField.Active:
+        return !todo.completed;
+      case FilterField.Completed:
+        return todo.completed;
+      case FilterField.All:
+      default:
+        return todo;
+    }
+  });
+
+  if (filter.query) {
+    newTodos = newTodos
+      .filter(({ title }) => title.toLowerCase()
+        .includes(filter.query.toLowerCase()));
+  }
+
+  return newTodos;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [choseTodoModal, setСhoseTodoModal] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Filter>({ field: FilterField.All, query: '' });
+  const [loading, setLoading] = useState(true);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(true);
```

Reply:
By default loader should be `false`, you need to set it to `true` only before getting data